### DISCUSSION TITLE
Help with new integration (newbee)

### DISCUSSION THREAD

**Discussion by Pinojunk** [djansen1987/SAJeSolar#25] [Q&A] [answered]
Hi, 
I recently install a SAJ H1 inverter with a chint ddsu666 sec module, I am a total beginner with home assistant but loving the experience so far.
so I found this SAJ eSolar integration and been trying run it with very limited success. I have some good values but all mixed up, and the good one I am not able to configure properly.
Please help, I am shure is my lack of knowledge.
Thanks

**Comment by djansen1987** [comment]
Hi @Pinojunk ,

No problem, could you share your component setup (**_without_** username/password_

**Comment by Pinojunk** [comment, marked as answer]
Hi Daniel,
Thx for your help, I really appreciate your work, hope this is what you need, please request any further info.

This is how I wrote the /config/configuration.yaml
##########################################################

sensor:
  - platform: saj_esolar
    username: 
    password: 
    sensors: h1
    resources:
      - nowPower
      - runningState
      - todayElectricity
      - monthElectricity
      - yearElectricity
      - totalElectricity # Energy -> Solar production
      - lastUploadTime
      - totalPlantTreeNum
      - totalReduceCo2
      - todayAlarmNum
      - status
      - plantuid
      - currency
      - address
      - isOnline
      - devOnlineNum
      - selfUseRate
      - totalBuyElec # Energy -> Grid consumption
      - totalConsumpElec
      - totalSellElec # Energy -> Return to grid
      - chargeElec # Energy -> Home Battery Storage -> Energy going in to the battery (kWh)
      - dischargeElec # Energy -> Home Battery Storage -> Energy coming out of the battery (kWh)

      - pvElec
      - useElec
      - buyElec
      - sellElec
      - buyRate
      - sellRate
      - selfConsumedRate1
      - selfConsumedRate2
      - selfConsumedEnergy1
      - selfConsumedEnergy2
      - batCapcity
      - batCurr
      - batEnergyPercent
      - batteryDirection
      - batteryPower
      - gridDirection
      - gridPower
      - h1Online
      - outPower
      - outPutDirection
      - pvDirection
      - pvPower
      - solarPower
      - totalLoadPower
      ##########################################################
![Energy card](https://user-images.githubusercontent.com/104317028/167246036-15f7ce78-1027-49f6-8071-b3e5e206878b.jpg)



![Developer Tools –  Esolar Home Assistant-1](https://user-images.githubusercontent.com/104317028/167246103-8e983139-d178-4677-985a-316ae9a7040d.jpg)

![Developer Tools –  Esolar Home Assistant-2](https://user-images.githubusercontent.com/104317028/167246104-11716017-c03f-499d-a758-a3e91e28e447.jpg)

![Developer Tools –  Esolar Home Assistant-3](https://user-images.githubusercontent.com/104317028/167246105-ed5f946a-a2ba-4a8c-8c4a-f75b4016aca6.jpg)

![Developer Tools –  Esolar Home Assistant-4](https://user-images.githubusercontent.com/104317028/167246106-c80f4b2f-259d-4e57-a252-213c4af884e5.jpg)

![Developer Tools –  Esolar Home Assistant-5](https://user-images.githubusercontent.com/104317028/167246101-4701e89b-1a79-4290-8103-30a8ae24f2ff.jpg)

**Reply by djansen1987** [reply]
HI @Pinojunk ,

Thanks for sharing. I can see you have some unavalible values. This is due to old entities. YOu proberbly tried adding the component by the first option, this has and resource called "todayGridIncome". Later you added the H1 sensor type to the config, this has a other set of resources. Old ones are kept by home assistant and have to be manualy deleted. In other words the H1 Component does not have the "todayGridIncome" resource.

in Youre case you have an H1 and a SEC module, i have never tested a setup like this so not sure if you could add both options at the same time in my component.

Only use the avalible resources linked to your setup.
In your case only the H1 part or SEC part

https://github.com/djansen1987/SAJeSolar#usage

**Reply by Pinojunk** [reply]
Thanks Daniel for you help, I will manually clean it.

**Reply by petakpa1** [reply]
Hi Daniel,

Could you help me too with correct configuring the configuration.yaml file

Currently it looks like as follows
sensor:
  - platform: saj_esolar
    username: _my user name used in eSolar app_
    password: _my pwd used in eSolar app_
    sensors: h1
    resources:
      - nowPower
      - runningState
      - todayElectricity
      - monthElectricity
      - yearElectricity
      - totalElectricity
      - todayGridIncome
      - income
      - lastUploadTime
      - totalPlantTreeNum
      - totalReduceCo2
      - todayAlarmNum
      - isAlarm
      - status
      - plantuid
      - currency
      - address
      - isOnline
      - peakPower
      - systemPower

Unfortunatelly I have got the following error measage:

Invalid config for [sensor.saj_esolar]: value must be one of ['address', 'batCapcity', 'batCurr', 'batEnergyPercent', 'batteryDirection', 'batteryPower', 'buyElec', 'buyRate', 'chargeElec', 'currency', 'devOnlineNum', 'dischargeElec', 'gridDirection', 'gridPower', 'h1Online', 'income', 'isAlarm', 'isOnline', 'lastUploadTime', 'monthElectricity', 'nowPower', 'outPower', 'outPutDirection', 'peakPower', 'plantTreeNum', 'plantname', 'plantuid', 'pvDirection', 'pvElec', 'pvPower', 'reduceCo2', 'runningState', 'selfConsumedEnergy1', 'selfConsumedEnergy2', 'selfConsumedRate1', 'selfConsumedRate2', 'selfUseRate', 'sellElec', 'sellRate', 'solarPower', 'status', 'todayAlarmNum', 'todayElectricity', 'todayGridIncome', 'totalBuyElec', 'totalBuyEnergy', 'totalConsumpElec', 'totalElectricity', 'totalGridPower', 'totalLoadEnergy', 'totalLoadPower', 'totalPlantTreeNum', 'totalPvEnergy', 'totalPvgenPower', 'totalReduceCo2', 'totalSellElec', 'totalSellEnergy', 'useElec', 'yearElectricity'] @ data['resources'][19]. Got 'systemPower'. (See ?, line ?). 

What is wrong with my config?

Thx in advance!
